### PR TITLE
Default style setting

### LINKED ISSUE #1876: Register the Style in the townhall (of the townhall) and make it as a default
Register the Style in the townhall (of the townhall) and make it as a default

Title says it all.

Needs a variable in the colony which is set on townhall placement (each townhall placement)

And this is set and chosen as default style if building is placed without buildingTool.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1876

# Changes proposed in this pull request:
- Store the townhall style in the colony and adapt it as default style to the buildings.
- Fix bug double adding colony buildings (especially for townhall)
- Fix bug with armor stands (Ignore all air blocks which might get returned by containers).

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import com.minecolonies.api.colony.requestsystem.requester.IRequester;
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutTownHall.java [hunk 2]
```diff
@@ -2,13 +2,16 @@
 
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.constant.Constants;
 import com.minecolonies.coremod.colony.Colony;
 import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.tileentities.TileEntityColonyBuilding;
 import net.minecraft.block.Block;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutTownHall.java]: MINOR Remove this unused import 'net.minecraft.block.Block'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 74 issues
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java#](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#): This file has 1.384 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L61](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L61): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 51 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L61](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L61): class "Colony" has 102 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L351](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L351): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L531](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L531): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L653](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L653): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1104](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1104): The Cyclomatic Complexity of this method "onWorldTick" is 20 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1547](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1547): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1547](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1547): Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1547](https://github.com/Minecolonies/minecolonies/blob/4fb91349d08fdb7ed3e2fdbf17048275b7c1ee75/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1547): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 71 issues
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 50 minor
* ![INFO][INFO] 4 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java#](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#): This file has 1.383 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L60](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L60): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 51 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L60](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L60): class "Colony" has 102 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L350](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L350): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L530](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L530): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L652](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L652): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1103](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1103): The Cyclomatic Complexity of this method "onWorldTick" is 20 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1546](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1546): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1546](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1546): Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1546](https://github.com/Minecolonies/minecolonies/blob/f6b33e6ad9d5ccfc8034ed8324730d48d0ba39dc/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1546): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**